templates/netlify/package.json

```
@@ -12,17 +12,18 @@
   "dependencies": {
     "@netlify/functions": "^1.0.0",
     "@remix-run/netlify": "*",
+    "@remix-run/netlify-edge": "*",
     "@remix-run/node": "*",
     "@remix-run/react": "*",
     "cross-env": "^7.0.3",
-    "react": "^17.0.2",
-    "react-dom": "^17.0.2"
+    "react": "^18.0.0",
+    "react-dom": "^18.0.0"
```

We're still using React v17 in our `templates`

```suggestion
    "react": "^17.0.2",
    "react-dom": "^17.0.2"
```

---

We're using React 18, because we're using streaming in the template. cc: @ascorbic

```jsx
import type { EntryContext } from "@remix-run/deno";
import { RemixServer } from "@remix-run/react";
import { renderToReadableStream } from "react-dom/server";
import isbot from "isbot";

export default async function handleRequest(
  request: Request,
  responseStatusCode: number,
  responseHeaders: Headers,
  remixContext: EntryContext
) {
  let body = await renderToReadableStream(
    <RemixServer context={remixContext} url={request.url} />,
    {
      onError() {
        responseStatusCode = 500;
      },
    }
  );

  if (isbot(request.headers.get("user-agent"))) {
    await body.allReady;
  }

  responseHeaders.set("Content-Type", "text/html");

  return new Response(body, {
    status: responseStatusCode,
    headers: responseHeaders,
  });
}
```

https://github.com/nickytonline/remix/blob/4f14a41c76d353598dae00900897e4e60815fc4d/templates/netlify/remix.init/app/entry.server.tsx#L3

---

Could we rewrite this in a v17 way for now?

All other templates are using v17, so I think we want to have this consistent.

Once we update to v18, we'll of course change this back to how it implemented currently.

---

I reverted it to v17 to keep it uniform, but also because of #2947.